main.py

```diff
@@ -15,39 +19,133 @@ def home():
 def selfie():
     return send_file("pictures/screenshot.jpeg", mimetype="image/jpeg")
 
-
-@app.route("/recipies", methods=["GET"])
-def recipies():
-    cippie_db = sqlite3.connect("cippedb.db")
-    cursor = cippie_db.cursor()
-    recipe = cursor.execute("SELECT food_name FROM recipies")
-    return jsonify({"status": "success", "data": recipe.fetchall()})
+    
+@app.route("/recipies", methods=["GET", "POST"])
+def get_recipies():
+    if request.method == "POST":
+        data = request.get_json()
+        try:
+            recipie = Recipie(food_name = data["name"], description = data["description"], top_image = data["top image"], bottom_image = data["bottom image"])
+            recipie.save()
+            return jsonify({"status": "success", "data": data}), 201
+        except:
+            return jsonify({"status": "error", "error": {"Important keys(in this order)": ["name", "description", "top image", "bottom image"]}}), 404
+    query = Recipie.select().dicts()
+    all_recipies = []
+    for item in query:
+        all_recipies.append(item)
+    return jsonify({"status": "success", "data": all_recipies})
 
 
 @app.route("/recipies/<name>", methods=["GET"])
 def get_recipie(name):
-    cippie_db = sqlite3.connect("cippedb.db")
-    cursor = cippie_db.cursor()
-    recipies = cursor.execute(
-        "SELECT food_name, description, top_image, bottom_image FROM recipies WHERE food_name=? ORDER BY food_name ASC LIMIT 1", 
-        (name, ))
-    return jsonify({"status": "success", "data": recipies.fetchall()})
+    query = Recipie.select().where(Recipie.food_name == name).tuples()
+    one_recipie = []
+    for item in query:
+        one_recipie.append(item)
+    return jsonify({"status": "success", "data": one_recipie})
+
 
+@app.route("/recipies/<idnumber>", methods=["PUT", "DELETE"])
+def update_recipie(idnumber):
+    if request.method == "PUT":
+        data = request.get_json()
+        update_data = Recipie.update(food_name = data["name"], description = data["description"], top_image = data["top image"], bottom_image = data["bottom image"].where(Recipie.id == idnumber))
+        update_data.execute()
+        query = Recipie.select().where(Recipie.id == idnumber).tuples()
+        updated_recipie = []
+        for item in query:
+            updated_recipie.append(item)
+        return jsonify({"status": "success", "data": updated_recipie})
+    elif request.method == "DELETE":
+        recipie = Recipie.get(Recipie.id == idnumber)
+        recipie.delete_instance()
+        return Recipie.get_or_none(Recipie.id == idnumber)
 
-@app.route("/recipies/<name>/<act>", methods=["GET"])
+
+@app.route("/recipies/<name>/<act>", methods=["GET", "POST"])
 def get_act(name, act):
-    cippe_db = sqlite3.connect("cippedb.db")
-    cursor = cippe_db.cursor()
-    key = cursor.execute("SELECT id FROM recipies WHERE food_name=?", (name, ))
-    key_id = key.fetchone()[0]
+    key = Recipie.get(Recipie.food_name == name).id
```

Comment:
I think this check `if request.method == "POST":` should come first

Reply:
so POST before checking act(ingredients or steps)? @jirevwe

Reply:
Yep

Reply:
Alright, done